In [69]:
import pandas as pd

df=pd.read_csv("/Users/cyrilmahfouz/Desktop/SEF/Week2/kc_house_data.csv")
df.describe().round()

id      price  bedrooms  bathrooms  sqft_living   sqft_lot  \
count  2.161300e+04    21613.0   21613.0    21613.0      21613.0    21613.0   
mean   4.580302e+09   540182.0       3.0        2.0       2080.0    15107.0   
std    2.876566e+09   367362.0       1.0        1.0        918.0    41421.0   
min    1.000102e+06    75000.0       0.0        0.0        290.0      520.0   
25%    2.123049e+09   321950.0       3.0        2.0       1427.0     5040.0   
50%    3.904930e+09   450000.0       3.0        2.0       1910.0     7618.0   
75%    7.308900e+09   645000.0       4.0        2.0       2550.0    10688.0   
max    9.900000e+09  7700000.0      33.0        8.0      13540.0  1651359.0   

        floors  waterfront     view  condition    grade  sqft_above  \
count  21613.0     21613.0  21613.0    21613.0  21613.0     21613.0   
mean       1.0         0.0      0.0        3.0      8.0      1788.0   
std        1.0         0.0      1.0        1.0      1.0       828.0   
min        1.0         0.0      0.0        1.0      1.0       290.0   
25%        1.0         0.0      0.0        3.0      7.0      1190.0   
50%        2.0         0.0      0.0        3.0      7.0      1560.0   
75%        2.0         0.0      0.0        4.0      8.0      2210.0   
max        4.0         1.0      4.0        5.0     13.0      9410.0   

       sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
count        21613.0   21613.0       21613.0  21613.0  21613.0  21613.0   
mean           292.0    1971.0          84.0  98078.0     48.0   -122.0   
std            443.0      29.0         402.0     54.0      0.0      0.0   
min              0.0    1900.0           0.0  98001.0     47.0   -123.0   
25%              0.0    1951.0           0.0  98033.0     47.0   -122.0   
50%              0.0    1975.0           0.0  98065.0     48.0   -122.0   
75%            560.0    1997.0           0.0  98118.0     48.0   -122.0   
max           4820.0    2015.0        2015.0  98199.0     48.0   -121.0   

       sqft_living15  sqft_lot15  
count        21613.0     21613.0  
mean          1987.0     12768.0  
std            685.0     27304.0  
min            399.0       651.0  
25%           1490.0      5100.0  
50%           1840.0      7620.0  
75%           2360.0     10083.0  
max           6210.0    871200.0

In [7]:
lat=df.lat.mean()
long=df.long.mean()

print("avrg lat of houses: ",lat)
print("avrg long of houses: ",long)

print(len(df))

avrg_area=df.sqft_lot.mean()+df.sqft_living.mean()
print("the avrg area of the houses: ",avrg_area)

print("the avrg building year of the houses: ",df.yr_built.mean())



avrg lat of houses:  47.56005251931708
avrg long of houses:  -122.21389640494147
21613
the avrg area of the houses:  17186.867302086706
the avrg building year of the houses:  1971.0051357978994


To start,we are looking at a dataframe representing house sales in King County, USA from Kaggle. To start this dataframe contains information about 21613 houses.We can see that these house revolves around a latitude of 47.56 and a longitude of -122.21.Also by calculating the averge of the sqft living and the sqft lot we can see that the average area of these houses is 17186.86 meter squarred and the average houses are built in the year 1971

In [35]:
built_renovated=df[['yr_built','yr_renovated','price']]
built_renovated.loc[built_renovated['yr_renovated']>0 ,'yr_renovated' ]=1

built_ren=built_renovated[['yr_built','yr_renovated','price']].groupby('yr_renovated').agg({'yr_built': 'count', 'price': 'mean'})

perc=(built_ren.iloc[1,0]/len(df))*100
print(built_ren)
print(f"the percentage of houses reovated is: {perc: .2f}%")

              yr_built          price
yr_renovated                         
0                20699  530447.958597
1                  914  760628.777899
the percentage of houses reovated is:  4.23%


Now, we notice that even if the average houses are built in the year 1977 only 914 of the houses are renovated which is equal to only 4.23% of all the houses. And we can notice that the prices of the houses renovated are more expencive that the old houses.<br>
<br>
Lets check now the variable that coukd possibly impact the price of the houses

In [50]:
price_waterfront=df[['price','waterfront']].groupby('waterfront').mean().round()
print("the prices according to the waterfonts")
print(price_waterfront)




the prices according to the waterfonts
                price
waterfront           
0            531653.0
1           1662524.0


Let's start with the houses that has waterfonts.<br>
We can remark that by grouping the houses that have waterfronts,that the values of the buildings that contains this feature are more expensive than the houses that aren't equipped with one.<br>
We did this by grouping the houses into 2 categories. The first one, the group that doesn't contain a waterfall and the second group contains a waterfall.Then we preceeded by getting the average price of these 2 groups and that's how we came up with this conclusion

In [52]:
price_conditions=df[['price','grade']]
sorted_price_up=price_conditions.sort_values(by='price',ascending=False)
print(sorted_price_up.head(10))

print(" ")
sorted_price_down=price_conditions.sort_values(by='price',ascending=True)
print(sorted_price_down.head(10))


          price  grade
7252  7700000.0     13
3914  7060000.0     11
9254  6890000.0     13
4411  5570000.0     13
1448  5350000.0     12
1315  5300000.0     12
1164  5110000.0     12
8092  4670000.0     12
2626  4500000.0     12
8638  4490000.0     12
 
         price  grade
1149   75000.0      3
15293  78000.0      5
465    80000.0      4
16198  81000.0      5
8274   82000.0      6
2141   82500.0      5
18468  83000.0      5
3767   84000.0      6
16714  85000.0      5
10253  85000.0      6


Now let's tackle the correlation between the price and the grade that the house is gived.<br>
In the first table, we can clearly see that the most expensive houses are attributed to the high graded houses.Similarly the least expensive houses have the lower grades amoung the buildings. <br>
Now how can someone give a grade?What are the conditions to a higher grade?

In [59]:
grade_area=df[['grade','sqft_living','sqft_lot']].groupby('grade').mean().round()
grade_area['avrg_area']=(grade_area['sqft_living']+grade_area['sqft_lot'])

print(grade_area.sort_values(by='avrg_area',ascending=False).head(4))

       sqft_living  sqft_lot  avrg_area
grade                                  
12          5472.0   54521.0    59993.0
11          4395.0   38373.0    42768.0
13          7483.0   28205.0    35688.0
10          3520.0   28191.0    31711.0


The average area of these houses are the largest houses on this dataframe.So one factor of the that give a house a high grade is the area of the house.

In [65]:
area_floors=df[['floors','sqft_living','sqft_lot']].groupby('floors').mean().round().head(4)
area_floors['avrg_area']=(area_floors['sqft_living']+area_floors['sqft_lot'])
print(area_floors)

        sqft_living  sqft_lot  avrg_area
floors                                  
1.0          1732.0   14568.0    16300.0
1.5          1908.0   16966.0    18874.0
2.0          2571.0   16134.0    18705.0
2.5          3145.0   17347.0    20492.0


We can see that the average area is increasing with the number of floors.The more floors a house has the bigger the area get

In [71]:
#where are located the most expensive houses
max_price_loc=df[['price','lat','long']]
sorted_price_up=max_price_loc.sort_values(by='price',ascending=False)
print(sorted_price_up.head(5))


          price      lat     long
7252  7700000.0  47.6298 -122.323
3914  7060000.0  47.6500 -122.214
9254  6890000.0  47.6305 -122.240
4411  5570000.0  47.6289 -122.233
1448  5350000.0  47.6232 -122.220


We can extract from this information that the most expensive houses are located around the same area.So we can conclude that this location is more fruitfull than others so maybe it could be a good place to invest in the future

<h3>BONUS</h3>